public/api/getPortfolioActivityView.php

```diff
@@ -46,40 +51,54 @@
         "isUserPortfolio" => is_null($row["portfolioCourseForUserId"]) ? "0" : "1",
         "courseLabel" => $row['courseLabel'],
         "courseImage" => $row['image'],
-        "courseColor" => $row['color']
+        "courseColor" => $row['color'],
+        "firstName" => $row['firstName'],
+        "lastName" => $row['lastName'],
+        "profilePicture" => $row['profilePicture'],
     ]);
     
 
     }else{
         throw new Exception("Activity not found.");
     }
 
-    //Add on the first and last name for each user portfolio
-    foreach($contributors as &$contributor){
-        if ($contributor['isUserPortfolio'] == '1'){
-         $courseId = $contributor['courseId'];
-         $sql = "
-            SELECT u.firstName,
-            u.lastName,
-            u.profilePicture
-            FROM user AS u
-            LEFT JOIN course AS c
-            ON u.userId = c.portfolioCourseForUserId
-            WHERE c.courseId = '$courseId'
-            ";
-        $result = $conn->query($sql);
+    //Add contributor history
+    $sql = "
+    SELECT
+    cch.courseId,
+    cch.isUserPortfolio,
+    c.label as courseLabel,
+    c.image,
+    c.color,
+    u.firstName,
+    u.lastName,
+    u.profilePicture
+    FROM content_contributor_history AS cch
+    LEFT JOIN course AS c
+        ON c.courseId = cch.courseId
+    LEFT JOIN user As u
+        ON u.userId = c.portfolioCourseForUserId
+    WHERE cch.doenetId = '$doenetId'
+    ORDER BY cch.timestamp DESC
+    ";
+    $result = $conn->query($sql);
 
-        if ($result->num_rows > 0) {
-            $row = $result->fetch_assoc();
-            $contributor['firstName'] = $row['firstName'];
-            $contributor['lastName'] = $row['lastName'];
-            $contributor['profilePicture'] = $row['profilePicture'];
-        }
+    if ($result->num_rows > 0) {
+        while($row = $result->fetch_assoc()){
+            array_push($contributors, [
+                "courseId" => $row["courseId"],
```

Comment:
fetch_assoc() already gives an associative array, we really don't have to keep doing this. You can control what you want out of the table using just the select list in the SQL.